packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -368,6 +368,10 @@ function hydrateCustomElement(
     vnode: VCustomElement,
     renderer: RendererAPI
 ): Node | null {
+    if (elm.hasAttribute('data-lwc-skip-hydrate')) {
+        return elm;
+    }
+    
```

Comment:
This skips hydration but I'm not sure what happens if the component is updated afterwards.

For example, if the parent has an update that would normally trigger VDOM diffing which would affect the child (e.g. removing it, passing in a new prop, etc.), LWC is not aware of the child, and rendering/callbacks will not be invoked.

Ideally we should have tests in `test-hydration` to verify this. It might work as-is (the element is just treated opaquely), but I'm not sure.

The custom element tag name may also not be defined, which could cause potential issues if other code is able to "hijack" it – e.g. someone injects code that does `customElements.define('lightning-button', ...)`. Not sure if this is a legitimate attack vector but it's at least unexpected.